src/modules/layout/containers/PermissionSnackbarContainer.js

```diff
@@ -0,0 +1,201 @@
+    const { locationPermissionStatus } = this.state
+    if (locationPermissionStatus === RESULTS.BLOCKED) {
+      await openSettings()
+    } else {
+      const result = await request(Platform.OS === 'ios'
+        ? PERMISSIONS.IOS.LOCATION_WHEN_IN_USE
+        : PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION)
+
+      if (result === RESULTS.GRANTED) {
+        this.hide()
+      }
+    }
+    this.updateSettingsAndPermissions()
+  }
+
+  requestPushNotificationPermissions = async () => {
+    // TODO NATIVE-399 Really request push notification permissions
+  }
+
+  checkShowSnackbar = async () => {
+    const { settings, locationPermissionStatus, pushNotificationPermissionStatus, showSnackbar: prevShow } = this.state
+
+    const showLocationSnackbar = settings !== null && settings.proposeNearbyCities === true &&
+      [RESULTS.BLOCKED, RESULTS.DENIED].includes(locationPermissionStatus) &&
+      this.landingRoute()
+
+    const showPushNotificationSnackbar = settings !== null && settings.allowPushNotifications === true &&
+      [RESULTS.BLOCKED, RESULTS.DENIED].includes(pushNotificationPermissionStatus) &&
+      this.dashboardRoute()
+
+    const showSnackbar = showLocationSnackbar ? 'LOCATION' : showPushNotificationSnackbar ? 'PUSH_NOTIFICATION' : null
+
+    if (showSnackbar !== prevShow) {
+      if (showSnackbar === null) {
+        this.hide()
+      } else if (prevShow === null) {
+        await this.setState({ showSnackbar })
+        this.show()
+      } else {
+        this.hideAndShow(showSnackbar)
+      }
+    }
+  }
+
+  landingRoute = (): boolean => {
+    const { navigation } = this.props
+    const { index, routes } = navigation.state
+    const currentRoute = routes[index]
+
+    return currentRoute.key === 'Landing'
+  }
+
+  dashboardRoute = (): boolean => {
+    const { navigation } = this.props
+    const { index, routes } = navigation.state
+    const currentRoute = routes[index]
+
+    return currentRoute.key === 'CityContent' && currentRoute.routes[currentRoute.index]?.routeName === 'Dashboard'
+  }
+
+  render () {
+    const { theme, t } = this.props
+    const { showSnackbar } = this.state
+
+    if (showSnackbar === 'LOCATION') {
+      return <Snackbar positiveAction={{ label: t('grant'), onPress: this.requestLocationPermissions }}
+                       negativeAction={{ label: t('deactivate'), onPress: this.deactivateProposeNearbyCities }}
+                       message={t('locationPermissionMissing')} theme={theme} animatedValue={this._animatedValue} />
+    } else if (showSnackbar === 'PUSH_NOTIFICATION') {
+      return <Snackbar positiveAction={{ label: t('grant'), onPress: this.requestPushNotificationPermissions }}
+                       negativeAction={{ label: t('deactivate'), onPress: this.deactivateAllowPushNotifications }}
+                       message={t('pushNotificationPermissionMissing')} theme={theme}
+                       animatedValue={this._animatedValue} />
+    }
```

Comment:
Make this 3 components: The actual animated snackbar, the one controlling location permission and the one controlling pushnotifications

Reply:
Not that easy. To extract the animation logic, I'd had to use refs. And by splitting location and push notifications permission it would become more difficult to handle the hiding animation and the case that both snackbars have to be shown

Reply:
But I'm of course open for proposals :)